akka-projection-dynamodb/src/main/scala/akka/projection/dynamodb/internal/DynamoDBProjectionImpl.scala

```diff
@@ -356,6 +447,99 @@ private[projection] object DynamoDBProjectionImpl {
           }
         }
       }
+
+      private def replayIfPossible(source: String, originalEnvelope: Envelope)(
+          implicit ec: ExecutionContext,
+          system: ActorSystem[_]): Future[Boolean] = {
+        originalEnvelope match {
+          case originalEventEnvelope: EventEnvelope[Any @unchecked] if originalEventEnvelope.sequenceNr > 1 =>
+            sourceProvider match {
+              case provider: LoadEventsByPersistenceIdSourceProvider[Any @unchecked]
+                  if offsetStore.settings.replayOnRejectedSequenceNumbers =>
+                val persistenceId = originalEventEnvelope.persistenceId
+                offsetStore.storedSeqNr(persistenceId).flatMap { storedSeqNr =>
+                  val fromSeqNr = storedSeqNr + 1
+                  val toSeqNr = originalEventEnvelope.sequenceNr
+                  log.debug(
+                    s"{} Replaying events after rejected sequence number. PersistenceId [{}], replaying from seqNr [{}] to [{}].",
+                    offsetStore.logPrefix,
+                    persistenceId,
+                    fromSeqNr,
+                    toSeqNr)
+                  provider.currentEventsByPersistenceId(persistenceId, fromSeqNr, toSeqNr) match {
+                    case Some(querySource) =>
+                      querySource
+                        .mapAsync(1) { envelope =>
+                          import DynamoDBOffsetStore.Validation._
+                          offsetStore
+                            .validate(envelope)
+                            .flatMap {
+                              case Accepted =>
+                                // FIXME: should we be grouping the replayed envelopes? based on what?
+                                loadEnvelope(envelope.asInstanceOf[Envelope], sourceProvider).flatMap {
+                                  loadedEnvelope =>
+                                    val offset = extractOffsetPidSeqNr(sourceProvider, loadedEnvelope)
+                                    if (isFilteredEvent(loadedEnvelope)) {
+                                      offsetStore.saveOffset(offset)
+                                    } else {
+                                      delegate
+                                        .process(Seq(loadedEnvelope))
+                                        .flatMap { _ =>
+                                          offsetStore.saveOffset(offset)
+                                        }
+                                    }
+                                }
```

Comment:
Only the accepted processing for grouped is different from at-least-once, so the rest could probably be shared between these implementations at least.

But this is currently processing the envelopes one at a time. Could make sense to group the replayed envelopes, since it's a grouped handler, but not sure what we would base the grouping on as we don't have knowledge here of what that it is for the projection.